File: 20260519/exos/03_for_liste.py
# ============================================================
# GTA — braquages de la semaine
# ============================================================
# Franklin a enregistré le butin de chaque braquage (en milliers $).
# Le jeu doit afficher le total pour le classement hebdo.
#
# Lance : python3 03_for_liste.py
# ============================================================

butins = [15, 8, 22, 5, 30]
total_butin = 0

for i in butins: 
    total_butin += i

# Parcours butins et additionne chaque valeur dans total_butin.
#
# tour 1 : x = 15  →  total_butin = 0 + 15  = 15
# tour 2 : x = 8   →  total_butin = 15 + 8  = 23
# tour 3 : x = 22  →  total_butin = 23 + 22 = 45
# tour 4 : x = 5   →  total_butin = 45 + 5  = 50
# tour 5 : x = 30  →  total_butin = 50 + 30 = 80
#
# Résultat attendu : total_butin == 80
#
# Indice : boucle for + +=

# À toi :


# --- Vérification (ne pas modifier) ---
assert total_butin == 80, f"Obtenu : {total_butin}, attendu : 80"
print("✅ Correct !")

File: 20260717/exos/03_for_liste.lua
-- ============================================================
-- GTA — butin de braquage
-- ============================================================
-- Franklin vient de finir un braquage. Chaque sac contient une
-- somme. Le jeu doit additionner tout le butin.
--
-- Lance : lua 03_for_liste.lua
-- ============================================================

sacs = {1200, 800, 450, 2100}
total = 0

-- Parcours la table sacs et additionne chaque valeur dans total.
--
-- tour 1 : x = 1200  →  total = 0 + 1200 = 1200
-- tour 2 : x = 800   →  total = 1200 + 800 = 2000
-- tour 3 : x = 450   →  total = 2000 + 450 = 2450
-- tour 4 : x = 2100  →  total = 2450 + 2100 = 4550
--
-- Résultat attendu : total == 4550
--
-- Indice : for _, x in ipairs(...) do ... end
--          et total = total + x  (pas de += en Lua)

-- À toi :
for i, x in ipairs(sacs) do
    total = total + x
end

-- --- Vérification (ne pas modifier) ---
assert(total == 4550, "Obtenu : " .. total .. ", attendu : 4550")
print("✅ Correct !")
